pkgs/tools/admin/gixy/default.nix

```diff
@@ -34,30 +35,44 @@ python.pkgs.buildPythonApplication rec {
     sha256 = "14arz3fjidb8z37m08xcpih1391varj8s0v3gri79z3qb4zq5k6b";
   };
 
-  postPatch = ''
-    sed -ie '/argparse/d' setup.py
-  '';
+  patches = [
+    # Migrate tests to pytest
+    # https://github.com/yandex/gixy/pull/146
+    (fetchpatch2 {
+      url = "https://github.com/yandex/gixy/compare/6f68624a7540ee51316651bda656894dc14c9a3e...b1c6899b3733b619c244368f0121a01be028e8c2.patch";
+      hash = "sha256-6VUF2eQ2Haat/yk8I5qIXhHdG9zLQgEXJMLfe25OKEo=";
+    })
+  ];
+
+  build-system = [ python.pkgs.setuptools ];
 
-  propagatedBuildInputs = with python.pkgs; [
+  dependencies = with python.pkgs; [
     cached-property
     configargparse
     pyparsing
     jinja2
-    nose3
-    setuptools
     six
   ];
 
-  meta = with lib; {
+  nativeCheckInputs = [ python.pkgs.pytestCheckHook ];
+
+  pythonRemoveDeps = [ "argparse" ];
+
+  passthru = {
+    inherit (nginx.passthru) tests;
+  };
+
+  meta = {
     description = "Nginx configuration static analyzer";
     mainProgram = "gixy";
     longDescription = ''
       Gixy is a tool to analyze Nginx configuration.
       The main goal of Gixy is to prevent security misconfiguration and automate flaw detection.
     '';
     homepage = "https://github.com/yandex/gixy";
-    license = licenses.mpl20;
-    maintainers = [ maintainers.willibutz ];
-    platforms = platforms.unix;
+    sourceProvenance = [ lib.sourceTypes.fromSource ];
```

Comment:
source should be not set as otherwise we had a lot of code deduplication

Reply:
See https://github.com/NixOS/nixpkgs/pull/329470#discussion_r1688547936 for my (and others’) thoughts on that. (If `sourceProvenance = [ lib.sourceTypes.fromSource ];` is redundant, so is `sourceProvenance = [ lib.sourceTypes.fromSource lib.sourceTypes.binaryNativeCode ];` and arguably even `sourceProvenance = [ lib.sourceTypes.binaryNativeCode ];`, not to mention `license = lib.licenses.free;` which nobody would argue for dropping.)